libs/common/src/sql-db-entities/base-entities/auditable.base-entity.ts

```diff
@@ -1,13 +1,39 @@
-import { Column, CreateDateColumn, UpdateDateColumn } from 'typeorm';
-import { DbAuditUpdatedByUserId } from '../helpers';
+import { Column, UpdateDateColumn } from 'typeorm';
+import { DbRequestSource } from '../helpers';
 
 export abstract class AuditableBaseEntity {
-  @CreateDateColumn()
-  createdDate!: Date;
-
   @UpdateDateColumn()
-  updatedDate!: Date;
+  lastUpdatedAt!: Date;
+
+  @Column({ type: 'nvarchar', nullable: true })
+  lastUpdatedByPortalUserId!: string | null;
+
+  @Column({ type: 'nvarchar', nullable: true })
+  lastUpdatedByTfmUserId!: string | null;
+
+  @Column()
+  lastUpdatedByIsSystemUser!: boolean;
 
-  @Column({ type: 'nvarchar' })
-  updatedByUserId!: DbAuditUpdatedByUserId;
+  public updateActivityDetails(requestSource: DbRequestSource): void {
```

Comment:
It's not a big deal, but I'm not the biggest fan of 'activity details' for the wording. I think it would be nicer if the name linked in some ways to the class name or the db field names. Perhaps 'updateLastUpdatedBy'?

Reply:
I used that name hoping someone would come up with a better suggestion, and I think `updateLastUpdatedBy` is good so I'll go with that. Thanks!